scripts/bootstrap.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+set -o pipefail
+set -o nounset
+set -o errexit
+
+err_report() {
+  echo "Exited with error on line $1"
+}
+trap 'err_report $LINENO' ERR
+
+IFS=$'\n\t'
+
+function grant_cluster_admin {
+  local username="$1"
+  kubectl create clusterrolebinding $username-cluster-admin-binding --clusterrole=cluster-admin --user=$username@gmail.com
+}
+
+function init_argocd {
+  kubectl create namespace argocd
+  kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
+
+  # TODO(giri): Should be exposed through ingress controller, doing this temporarily
+  kubectl patch svc argocd-server -n argocd -p '{"spec": {"type": "LoadBalancer"}}'
+}
+
+function download_argocd_tools() {
+  if [ "$(uname)" == "Darwin" ]; then
+    brew tap argoproj/tap
+    brew install argoproj/tap/argocd
+  elif [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]; then
+    VERSION=$(curl --silent "https://api.github.com/repos/argoproj/argo-cd/releases/latest" | grep '"tag_name"' | sed -E 's/.*"([^"]+)".*/\1/')
+    curl -sSL -o /usr/local/bin/argocd https://github.com/argoproj/argo-cd/releases/download/$VERSION/argocd-linux-amd64
+    chmod +x /usr/local/bin/argocd
+  fi
+}
+
+### main program
+init_argocd
+download_argocd_tools
+
+# TODO(admin): Anyone that wants to execute argocd need to be cluster admin
+grant_cluster_admin girikuncoro
```

Comment:
isn't it better to check if the username is admin or not? not only one specific person? any ideas?

Reply:
this is hard to check unless you know what resource/action do you really need and I haven't studied the GKE identity auth

cluster-admin is requirement from argocd to create new cluster roles, we can add new role for that gives this permission, but for now let's make it flexible until we learn enough

ref: https://github.com/argoproj/argo-cd/blob/master/docs/getting_started.md#1-install-argo-cd